src/unix/linux-core.c

```diff
@@ -1011,3 +1011,45 @@ uint64_t uv_get_total_memory(void) {
 
   return 0;
 }
+
+
+static uint64_t uv__read_cgroups_uint64(const char* cgroup, const char* param) {
+  char filename[256];
+  uint64_t rc;
+  int fd;
+  ssize_t n;
+  char buf[32];  /* Large enough to hold an encoded uint64_t. */
+
+  snprintf(filename, "/sys/fs/cgroup/%s/%s", cgroup, param);
+
+  rc = 0;
+  fd = uv__open_cloexec(filename, O_RDONLY);
+
+  if (fd < 0)
+    return 0;
+
+  n = read(fd, buf, sizeof(buf) - 1);
+
+  if (n > 0) {
+    buf[n] = '\0';
+    sscanf(buf, "%llu", &rc);
+  }
+
+  if (uv__close_nocheckstdio(fd))
+    abort();
+
+  return rc;
+}
+
+
+uint64_t uv_get_constrained_memory(void) {
+  uint64_t rc;
+
+  rc = uv__read_cgroups_uint64("memory", "memory.limit_in_bytes");
+
+  if (rc != 0)
+    return rc;
+
+  /* Usable memory is not constrained by cgroups. */
+  return 0;
```

Comment:
Lines 1050-1054 are somewhat tautological, you could just replace the body of this function with this :-)
```c
uint64_t uv_get_constrained_memory(void) {
  return uv__read_cgroups_uint64("memory", "memory.limit_in_bytes");
}
```

Reply:
Done, thanks for the pointer.